### ISSUE TITLE
(#613) consecutive install -g  doesn't bother getting data

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#862]
```shell
$> datalad install -d . -r labs
2016-09-20 08:17:09,548 [INFO   ] Installing /tmp/datasets.datalad.org/labs (install.py:434)
2016-09-20 08:17:09,548 [INFO   ] Install subdataset at: /tmp/datasets.datalad.org/labs (install.py:452)
2016-09-20 08:17:09,562 [INFO   ] Installing /tmp/datasets.datalad.org/labs from http://datasets.datalad.org/labs/.git (install.py:434)
2016-09-20 08:17:09,562 [INFO   ] Install dataset at: /tmp/datasets.datalad.org/labs (install.py:492)
2016-09-20 08:17:09,713 [INFO   ] Installing /tmp/datasets.datalad.org/labs/mitchell (install.py:434)
2016-09-20 08:17:09,713 [INFO   ] Install subdataset at: /tmp/datasets.datalad.org/labs/mitchell (install.py:452)
2016-09-20 08:17:09,726 [INFO   ] Installing /tmp/datasets.datalad.org/labs/mitchell from http://datasets.datalad.org/labs/mitchell/.git (install.py:434)
2016-09-20 08:17:09,726 [INFO   ] Install dataset at: /tmp/datasets.datalad.org/labs/mitchell (install.py:492)
2016-09-20 08:17:09,878 [INFO   ] Installing /tmp/datasets.datalad.org/labs/mitchell/science2008 (install.py:434)
2016-09-20 08:17:09,878 [INFO   ] Install subdataset at: /tmp/datasets.datalad.org/labs/mitchell/science2008 (install.py:452)
2016-09-20 08:17:09,892 [INFO   ] Installing /tmp/datasets.datalad.org/labs/mitchell/science2008 from http://datasets.datalad.org/labs/mitchell/science2008/.git (install.py:434)
2016-09-20 08:17:09,892 [INFO   ] Install dataset at: /tmp/datasets.datalad.org/labs/mitchell/science2008 (install.py:492)
2016-09-20 08:17:10,531 [INFO   ] Initializing annex repo at /tmp/datasets.datalad.org/labs/mitchell/science2008 (install.py:544)
2016-09-20 08:17:10,558 [INFO   ] Initializing annex repo at /tmp/datasets.datalad.org/labs/mitchell/science2008 (install.py:544)
2016-09-20 08:17:10,583 [INFO   ] Saving changes to <Dataset path=/tmp/datasets.datalad.org/labs/mitchell> (install.py:592)
2016-09-20 08:17:10,599 [INFO   ] Saving changes to <Dataset path=/tmp/datasets.datalad.org/labs> (install.py:592)
2016-09-20 08:17:10,621 [INFO   ] Saving changes to <Dataset path=/tmp/datasets.datalad.org> (install.py:592)
2016-09-20 08:17:10,670 [INFO   ] Nothing to save, consider auto-detection of changes, if this is unexpected. (save.py:246)
3 installed items are available at
<Dataset path=/tmp/datasets.datalad.org/labs>
<Dataset path=/tmp/datasets.datalad.org/labs/mitchell>
<Dataset path=/tmp/datasets.datalad.org/labs/mitchell/science2008>
changes on filesystem:                                                                                                                                  
 openfmri | 0

$> datalad install -d . -r -g labs
2016-09-20 08:17:19,078 [INFO   ] Installing /tmp/datasets.datalad.org/labs (install.py:434)
2016-09-20 08:17:19,078 [INFO   ] Install subdataset at: /tmp/datasets.datalad.org/labs (install.py:452)
2016-09-20 08:17:19,148 [INFO   ] Getting data of <Dataset path=/tmp/datasets.datalad.org/labs> (install.py:583)
2016-09-20 08:17:19,148 [INFO   ] Saving changes to <Dataset path=/tmp/datasets.datalad.org> (install.py:592)
2016-09-20 08:17:19,192 [INFO   ] Nothing to save, consider auto-detection of changes, if this is unexpected. (save.py:246)
1 installed item is available at
<Dataset path=/tmp/datasets.datalad.org/labs>
```

So that execution is independent of the fact either you ran install before or not, it should call get even if dataset is already there imho

**Comment by bpoldrack** [comment]
Indeed. `get` isn't called correctly in the recursive case. (Ignored because toplevel isn't an annex)

**Comment by yarikoptic** [comment]
should have got closed by #613 merge
